package.json

```diff
@@ -17,14 +17,16 @@
   },
   "type": "module",
   "devDependencies": {
+    "@eslint/js": "^9.32.0",
     "@testing-library/react": "^16.0.0",
     "@types/node": "^24.0.1",
     "@types/ws": "^8.5.10",
-    "eslint": "^9.0.0",
+    "eslint": "^9.32.0",
     "globals": "^16.0.0",
     "jsdoc": "^4.0.2",
     "jsdom": "^26.0.0",
-    "typescript": "^5.2.2",
+    "typescript": "^5.9.2",
+    "typescript-eslint": "^8.39.0",
```

Comment:
Sorry to keep being a stickler about PR separation of concerns - I promise I'll be responsive to reviewing this over the next few business days in return!

Could we do the typescript config changes and the eslint config changes in a separate PR before this one? They should be incrementally adoptable since we already are using both typescript and ESLint.

I mainly ask because the _great majority_ of the diff on this PR is the `package-lock.json`, lol.

Reply:
merged that one, thanks!

argh, I forgot this repo uses squash-merge - so you'll have to rebase this manually since the commit hash gets changed by merging. sorry about that!